File: frontend/src/api/fraud.js
import apiClient from './client';

const fraudService = {
    /**
     * Get fraud alerts
     * @param {boolean} resolved
     * @returns {Promise} List of alerts
     */
    async getAlerts(resolved = false) {
        const response = await apiClient.get(`/api/v1/fraud/?resolved=${resolved}`);
        return response.data;
    },

    /**
     * Resolve alert
     * @param {number} id
     * @returns {Promise}
     */
    async resolveAlert(id) {
        const response = await apiClient.put(`/api/v1/fraud/${id}/resolve`);
        return response.data;
    },
};

export default fraudService;

File: backend/app/api/v1/endpoints/fraud.py
from typing import List, Any
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy.future import select
from app.api import deps
from app.models.fraud_alert import FraudAlert
from app.models.user import User
from pydantic import BaseModel
from datetime import datetime

class FraudAlertResponse(BaseModel):
    id: int
    payment_id: int
    severity: str
    description: str
    is_resolved: bool
    detected_at: datetime
    
    class Config:
        from_attributes = True

router = APIRouter()

@router.get("/", response_model=List[FraudAlertResponse])
async def list_fraud_alerts(
    resolved: bool = False,
    db: AsyncSession = Depends(deps.get_db),
    current_user: User = Depends(deps.get_current_user),
) -> Any:
    """List fraud alerts"""
    query = select(FraudAlert).where(FraudAlert.is_resolved == resolved).order_by(FraudAlert.detected_at.desc())
    result = await db.execute(query)
    return result.scalars().all()

@router.put("/{alert_id}/resolve")
async def resolve_alert(
    alert_id: int,
    db: AsyncSession = Depends(deps.get_db),
    current_user: User = Depends(deps.get_current_user),
) -> Any:
    """Resolve a fraud alert"""
    result = await db.execute(select(FraudAlert).where(FraudAlert.id == alert_id))
    alert = result.scalars().first()
    if not alert:
        raise HTTPException(status_code=404, detail="Alert not found")
    
    alert.is_resolved = True
    alert.resolved_at = datetime.utcnow()
    alert.resolved_by = current_user.id
    
    await db.commit()
    return {"message": "Alert resolved"}
